oak/src/oak/server.cpp

```diff
@@ -22,26 +22,49 @@
 
 #include "oak/server.hpp"
 
+#include <chrono>
 #include <condition_variable>  // for condition_variable
-#include <functional>          // for bind
-#include <map>                 // for map<>
-#include <memory>              // for make_shared<>
-#include <mutex>               // for mutex, unique_lock
-#include <sstream>             // for stringstream
-#include <string>              // for string
-
+#include <ctime>
+#include <functional>  // for bind
+#include <iostream>
+#include <map>      // for map<>
+#include <memory>   // for make_shared<>
+#include <mutex>    // for mutex, unique_lock
+#include <sstream>  // for stringstream
+#include <string>   // for string
+#include <thread>   // for thread
+
+#include <boost/beast/core.hpp>  // for boost::beast
+#include <boost/beast/http.hpp>  // for boost::beast::http
+#include <boost/format.hpp>
 #include <boost/tokenizer.hpp>  // for tokenizer, char_separator
+#include <boost/version.hpp>    // for BOOST_VERSION
 
-// Requires: boost-wannabe cppnetlib
-#include <boost/network/include/http/server.hpp>
-#include <boost/network/utils/thread_pool.hpp>
+#include <cloe/core.hpp>        // for logger::get
+#include <cloe/handler.hpp>     // for Request
+#include "oak/requeststub.hpp"  // for RequestStub
+using namespace cloe;           // NOLINT(build/namespaces)
 
-#include <cloe/core.hpp>     // for logger::get
-#include <cloe/handler.hpp>  // for Request
-using namespace cloe;        // NOLINT(build/namespaces)
+// Alias namespaces
+namespace asio = boost::asio;
+namespace ip = boost::asio::ip;
+namespace beast = boost::beast;
+namespace http = boost::beast::http;
 
 namespace oak {
 
+// The type of the used socket
+using socket_type = ip::tcp::socket;
```

Comment:
I like the using statements here. Helps make this readable. However, I don't find the `_type` suffix helpful -- I spent half the PR thinking that it was describing the socket type, like HTTP or PIPE, or whatnot.

Elsewhere in the project, we just transform it into our style, so like `using Json = nlohmann::json`.

Or you can wrap it all in a short namespace, like:
```
namespace bb {
using flat_buffer = beast::flat_buffer;
}
```

Or you can change the name to `flat_buffer_t`. That's less confusing, but it's also very much C-style naming conventions, which is not very modern considering we're almost coding in C++17 now.